### PR TITLE
PAINTROID-434 An option to enable/disable and adjust Zoom

### PR DESCRIPTION
**Pull request by Lenkomotive**
Ticket: [PAINTROID-434](https://jira.catrob.at/browse/PAINTROID-434)

A dialog box that contains options for enabling and disabling the zoom. It will also contain a slider that can be used to adjust the level of zoom.

PS:
This is the original PR: https://github.com/Catrobat/Paintroid/pull/1155
Due to the amount of conflicts, I decided to do a manual merge onto the current develop branch.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x] Stick to the project’s gitflow workflow
- [x] Verify that your changes do not have any conflicts with the base branch
- [ ] After the PR, verify that all CI checks have passed
- [x] Post a message in the *#paintroid* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

**Comment by bakicelebi** [comment]
@Lenkomotive please resolve merge conflicts.

File: Paintroid/src/main/java/org/catrobat/paintroid/ui/zoomwindow/DefaultZoomWindowController.kt [hunk 1]
```diff
@@ -29,32 +30,30 @@ class DefaultZoomWindowController
...
 
     private val canvasRect = Rect()
     private val checkeredPattern = Paint()
     private val framePaint = Paint()
 
-    private val windowHeight =
-        activity.resources.getDimensionPixelSize(R.dimen.pocketpaint_zoom_window_height)
-    private val windowWidth =
-        activity.resources.getDimensionPixelSize(R.dimen.pocketpaint_zoom_window_width)
+    private val windowSideDimen = activity.resources.getDimensionPixelSize(R.dimen.pocketpaint_zoom_window_height)
```
Comment (bakicelebi) [inline Paintroid/src/main/java/org/catrobat/paintroid/ui/zoomwindow/DefaultZoomWindowController.kt:42]: You might need to use correct window dimensions `windowSideDimen = activity.resources.getDimensionPixelSize(R.dimen.pocketpaint_zoom_window_height)` is used as width later on

**Review by bakicelebi — COMMENTED** [review]
Minor changes needed see comments && resolve conflicts.

File: Paintroid/src/main/java/org/catrobat/paintroid/ui/zoomwindow/DefaultZoomWindowController.kt (hunk 1 above)
Comment (Lenkomotive) [reply Paintroid/src/main/java/org/catrobat/paintroid/ui/zoomwindow/DefaultZoomWindowController.kt:42]: I will take a look at it, thanks!

File: Paintroid/src/main/res/values/string.xml [hunk 2]
```diff
@@ -99,6 +99,7 @@
     <string name="menu_export">Export</string>
     <string name="menu_feedback">Feedback</string>
     <string name="menu_advanced">Advanced settings</string>
+    <string name="menu_zoom_settings">Zoom Window Settings</string>
```
Comment (bakicelebi) [inline Paintroid/src/main/res/values/string.xml:102]: For consistency sake first letter of the first word uppercase and rest lowercase: Zoom window settings

**Review by bakicelebi — APPROVED** [review]
:+1:

**Review by exland — APPROVED** [review]
LGTM

**Comment by ThorstenBandel** [comment]
There are more tests than normally failing. The following tests do not fail on develop. Could you take a look at those?

- testCheckmarkLineFeatureChangeShape
- testCheckmarkLineFeatureChangeStrokeWidth

**Comment by Lenkomotive** [comment]
>

**Review by ThorstenBandel — APPROVED** [review]
LGTM

**Pull request merged**